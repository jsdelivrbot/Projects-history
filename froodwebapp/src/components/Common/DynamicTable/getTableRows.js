/* eslint-disable babel/new-cap, react/no-array-index-key, react/jsx-filename-extension */
import React from 'react';
import { Checkbox } from 'antd';
import { Row, Cell } from './TableComponents';
import ConnectedDraggableHeader from './DraggableHeader';
import styles from './StickyTable.scss';

const getTableRows = (
  headers,
  data,
  selectedRows,
  handleRowClick,
  handleDownloadItem,
  handleSelectRows,
  handleSort,
  actionColumnVisible,
  moveHeader,
  saveHeaderOrder
) => {
  const rows = [];
  rows.push(
    <Row key="headers">
      <Cell>
        <Checkbox
          onClick={ e => handleSelectRows('all', e.target.checked) }
        />
      </Cell>
      { headers.map((header, headerIndex) => (
        <ConnectedDraggableHeader
          key={ header.key }
          index={ headerIndex }
          moveHeader={ moveHeader }
          saveHeaderOrder={ saveHeaderOrder }
          isSortable={ header.isSortable }
          handleSort={ handleSort }
          headerId={ header.id }
          headerText={ header.name }
          headerDataType={ header.dataType }
        />
      ))}
      { actionColumnVisible &&
        <Cell>
          Action
        </Cell>
      }
    </Row>
  );

  data.map((item, rowIndex) => (
    rows.push(
      <Row id={ item.id } key={ rowIndex } onClick={ handleRowClick }>
        <Cell>
          <Checkbox
            onClick={ () => handleSelectRows(item.id) }
            checked={ selectedRows.indexOf(item.id) !== -1 }
          />
        </Cell>
        {
          headers.map((header, headerIndex) => (
            <Cell
              key={ `${rowIndex}-${headerIndex}` }
              style={ { textAlign: header.dataType.includes('decimal') ? 'right' : 'left' } }
            >
              { (!header.isEditable
               && header.dataType.includes('decimal')
               && Number(item[header.dataAlias]).toFixed(2))
               || item[header.dataAlias]
              }
            </Cell>
          ))
        }
        { actionColumnVisible &&
          <Cell>
            <span>
              <a
                role="button"
                tabIndex="-1"
                id={ item.id }
                className={ styles.actionText }
                onClick={ handleDownloadItem }
              >
                Download
              </a>
            </span>
          </Cell>
        }
      </Row>
    )
  ));
  return rows;
};

export default getTableRows;
