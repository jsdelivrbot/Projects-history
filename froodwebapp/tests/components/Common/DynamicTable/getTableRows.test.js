import React from 'react';
import getTableRows from 'components/Common/DynamicTable/getTableRows';
import renderer from 'react-test-renderer';

const { describe, it, expect } = global;

describe('getTableRows', () => {
  it('captures snapshot of getTableRows', () => {
    const props = {
      selectedRows: ['first row', 'second row'],
      actionColumn: false,
      data: [{
        id: 'first row'
      }, {
        id: 'second row'
      }],
      headers: [{
        id: 2,
        value: 'some value',
        key: 2,
        moveHeader: jest.fn(),
        saveHeaderOrder: jest.fn(),
        isSortable: true,
        handleSort: jest.fn(),
        name: 'second header',
        dataType: 'number'
      }],
      handleSelectRows: jest.fn(),
      handleUpdateColumnsOrder: jest.fn(),
      handleSort: jest.fn(),
      handleRowClick: jest.fn(),
      handleDownloadItem: jest.fn()
    };
    const snappedComponent = renderer.create(<getTableRows { ...props } />).toJSON();

    expect(snappedComponent).toMatchSnapshot();
  });
});
