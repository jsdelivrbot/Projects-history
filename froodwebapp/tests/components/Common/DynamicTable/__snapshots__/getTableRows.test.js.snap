// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTableRows captures snapshot of getTableRows 1`] = `
<getTableRows
  actionColumn={false}
  data={
    Array [
      Object {
        "id": "first row",
      },
      Object {
        "id": "second row",
      },
    ]
  }
  handleDownloadItem={[Function]}
  handleRowClick={[Function]}
  handleSelectRows={[Function]}
  handleSort={[Function]}
  handleUpdateColumnsOrder={[Function]}
  headers={
    Array [
      Object {
        "dataType": "number",
        "handleSort": [Function],
        "id": 2,
        "isSortable": true,
        "key": 2,
        "moveHeader": [Function],
        "name": "second header",
        "saveHeaderOrder": [Function],
        "value": "some value",
      },
    ]
  }
  selectedRows={
    Array [
      "first row",
      "second row",
    ]
  }
/>
`;
